source/solvers/heat_transfer_assemblers.cc

```diff
@@ -786,24 +788,139 @@ HeatTransferAssemblerLaser<dim>::assemble_rhs(
               // respectively.
               local_rhs(i) += laser_heat_source * phi_T_i * JxW;
             }
+        } // end loop on quadrature points
+    }
+}
+
+template class HeatTransferAssemblerLaserExponentialDecay<2>;
+template class HeatTransferAssemblerLaserExponentialDecay<3>;
+
+template <int dim>
+void
+HeatTransferAssemblerLaserHeatFluxVOFInterface<dim>::assemble_matrix(
+  HeatTransferScratchData<dim> & /*scratch_data*/,
+  StabilizedMethodsCopyData & /*copy_data*/)
+{}
+
+template <int dim>
+void
+HeatTransferAssemblerLaserHeatFluxVOFInterface<dim>::assemble_rhs(
+  HeatTransferScratchData<dim> &scratch_data,
+  StabilizedMethodsCopyData    &copy_data)
+{
+  // Laser parameters
+  const double concentration_factor = laser_parameters->concentration_factor;
+  const double laser_power          = laser_parameters->laser_power;
+  const double absorptivity         = laser_parameters->laser_absorptivity;
+  const double laser_start_time     = laser_parameters->start_time;
+  const double laser_end_time       = laser_parameters->end_time;
+  const double beam_radius          = laser_parameters->beam_radius;
+  const double current_time = this->simulation_control->get_current_time();
+
+  if (current_time >= laser_start_time && current_time <= laser_end_time)
+    {
+      // Get laser path
+      Function<dim> &laser_scan_path = *(laser_parameters->laser_scan_path);
```

Comment:
This should be refactored later to be set outside of the assembler. It's not necessary to set the time at every cell...